Tools/Scripts/libraries/webkitbugspy/webkitbugspy/bugzilla.py

```diff
@@ -84,20 +91,35 @@ def from_string(self, string):
         return None
 
     def credentials(self, required=True):
-        return credentials(
+        return webkitcorepy.credentials(
             url=self.url,
             required=required,
             prompt=self.url.split('//')[-1],
         )
 
+    def _login_arguments(self, required=False, query=None):
+        username, password = self.credentials(required=required)
+        if not username or not password:
+            return '?{}'.format(query) if query else ''
+        return '?login={username}&password={password}{query}'.format(
+            username=username,
+            password=password,
+            query='&{}'.format(query) if query else '',
+        )
+
+    @decorators.Memoize()
```

Comment:
Nit: to keep consistency with `github.py`, maybe use `@webkitcorepy.decorators.Memoize()` instead?